### ISSUE TITLE
Queries being routed to server even though not all server segments are online yet

### ISSUE THREAD

**Issue by dang-stripe** [apache/pinot#9056]
Our offline segment servers take a very long time to load all segments. We've set a very high value for `pinot.server.startup.timeoutMs` to compensate for this and to avoid having the server receive queries until all segments have moved to online state.

We noticed that there are still queries being routed to the server while it's not yet healthy though. Does the broker only wait for the segments to be fully online and not check if the server is considered fully healthy yet? Is there a configuration that would prevent this? The server's resources are under heavy load during segment loads, so any queries sent during this time would fail.

The remaining pinot server conf is mostly using defaults. `pinot.server.startup.minResourcePercent` is not set, which I believe defaults to 100%.

```
[2022-07-13 19:22:48.586429] INFO [PinotServiceManager] [Start a Pinot [SERVER]] Trying to start Pinot Server...

// seeing this server in broker logs at with failure to respond (responseTimeMs == -1):
2022-07-13 22:20:44.229
2022-07-13 22:32:30.344
2022-07-13 22:47:29.760
...

[2022-07-13 23:11:04.748064] INFO [BaseServerStarter] [Start a Pinot [SERVER]] Service status is GOOD after 13683066ms
```

**Comment by dang-stripe** [comment]
This is a newly added server. We don't have those fixes in our version so I'll try to pull them in. Which helix callback does this happen in just so I know what to search for or is there a metric for this?

**Comment by mcvsubbu** [comment]
There is a heatlhcheck call on the server that returns good only if all segments for the server are loaded and available in external view.

**Comment by Jackie-Jiang** [comment]
Sync'ed offline, and the root cause of the issue is that the server is new added, so broker will start routing queries to it once the segments become online in external view. #8525 should be able to fix this

**Comment by suddendust** [comment]
We've seen this too in 0.9.x. Queries get routed to unavailable brokers.

**Comment by Jackie-Jiang** [comment]
@suddendust Just to clarify, do you main queries get routed to unavailable servers? The fix in #8525 will be available in the next release (0.11.0)
